src/Packages/java/lang/_String.php

```diff
@@ -344,14 +346,32 @@ public function indexOf($a = null, $b = null)
     /**
      * Returns a canonical representation for the string object.
      *
-     * @param mixed $a
+     * @native ConstantPool
+     * @param ConstantPool $cp
      * @return mixed
-     * @throws NotImplementedException
+     * @throws ReadOnlyException
      * @see https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/package-summary.html#intern
      */
-    public function intern($a = null)
-    {
-        throw new NotImplementedException(__METHOD__);
+    public function intern(ConstantPool $cp)
+    {
+        // Find same string in the Constant Pool
+        foreach ($cp as $key => $value) {
+            if (!($value instanceof _Utf8)) {
+                continue;
+            }
+            /**
+             * @var _Utf8 $value
+             */
+            if ((string) $value === (string) $this->object) {
+                $this->object = $value
+                    ->enableWrite(true)
+                    ->setString((string) $this->object)
+                    ->setStringObject($this)
+                    ->enableWrite(false);
```

Comment:
Because `$this->object` may be a string (on PHP type) or an other obejct.
`$this->object` allows dynamic generated object, not just referred the Constant Pool.

For the above reasons, If the code substitutes a value to `$this-object`, `$this-object` ensures the Constant Pool entry string data. (In this case, it ensures a `_Utf8` structure.)

Herewith, If `operand stack` is stacking this object, but operations can be proceeded correctly.

Reply:
@chitoku-k 
Because $this->object may be a string (on PHP type) or an other obejct.
$this->object allows dynamic generated object, not just referred the Constant Pool.

For the above reasons, If the code substitutes a value to $this-object, $this-object ensures the Constant Pool entry string data. (In this case, it ensures a _Utf8 structure.)

Herewith, If operand stack is stacking this object, but operations can be proceeded correctly.

Reply:
Finally, it turned out that `$this->object` must be equal to the instance already existing in the constant pool because any later comparison might not be guaranteed to treat them equal (such as `if_a` operation). Although it somewhat looks like `String#intern()` is affected by the internal object in the constant pool, this implementation is actually derived from the restriction of the way `_Utf8` and `_String` relate to each other.